### PR TITLE
WavelengthCoverage

### PR DESCRIPTION
**Pull request by rpiaggio**
Introduces a `WavelengthCoverage` model class, which represents a wavelength width. It can be centered around (or starting at, or ending at) a `Wavelength` to obtain a `BoundedInterval[Wavelength]` representing coverage.

Replaces previous `Coverage` class.

(Maybe we should name this `WavelengthWidth`??)

### CODE REVIEW

**Review by swalker2m — APPROVED** [review]
Looks great! :+1:

**Comment by swalker2m** [comment]
> Maybe we should name this WavelengthWidth??

Coverage seems to be the term the astronomers use.  I don't know; it seems good to me.

**Comment by rpiaggio** [comment]
I was wondering if we should distinguish between the terms for the range (ie: `[λ1, λ2]`) and the width of the range (ie: `λ2 - λ1`; `-` is a minus here). We seem to be using "coverage" interchangeably for both.

File: modules/core/shared/src/main/scala/lucuma/core/math/WavelengthCoverage.scala [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+     * Alias for `toPicometers`.
+     */
+    def pm: Quantity[PosInt, Picometer] =
+      toPicometers
+
+    /**
+      * Create a wavelength range centered on the given value. Will be truncated at 
+      * Wavelength.Min and Wavelength.Max.
+      */
+    def centeredAt(λ: Wavelength): BoundedInterval[Wavelength] =
```
Comment (toddburnside) [inline modules/core/shared/src/main/scala/lucuma/core/math/WavelengthCoverage.scala:50]: I assume planning to use this class to calculate `coverageInterval` in the SpectroscopyModesMatrix in explore? It seems there we have another case where the interval gets `clipped` within a larger interval. So given a central wavelength, the coverage interval can't extend beyond the capabilities of the configuration. Do you want to include that case here, or just handle it manually in SpectroscopyModesMatrix?

**Review by toddburnside — APPROVED** [review]
LGTM

**Comment by toddburnside** [comment]
> I was wondering if we should distinguish between the terms for the range (ie: `[λ1, λ2]`) and the width of the range (ie: `λ2 - λ1`; `-` is a minus here). We seem to be using "coverage" interchangeably for both.

I see what you mean. Maybe `CoverageWidth` or `WavelengthCoverageWidth`?

**Comment by swalker2m** [comment]
> I see what you mean. Maybe `CoverageWidth` or `WavelengthCoverageWidth`?

At least in one version of the inception document it seems we label this "λ Range" (in the configuration panel) and use "λ Coverage" for the interval itself (in advanced configuration) so maybe `WavelengthRange`?  If we're using a `BoundedInterval[Wavelength]` for the coverage itself then the "coverage" term goes away in the type names I guess.

**Comment by rpiaggio** [comment]
Ok, renaming to `WavelengthRange` since it seems consistent with the inception document.

**Comment by rpiaggio** [comment]
Ooops merged before pushing the rename commit. I'll send another PR.

File: modules/core/shared/src/main/scala/lucuma/core/math/WavelengthCoverage.scala (hunk 1 above)
Comment (rpiaggio) [reply modules/core/shared/src/main/scala/lucuma/core/math/WavelengthCoverage.scala:50]: > Do you want to include that case here, or just handle it manually in SpectroscopyModesMatrix?

It's done via `BoundedInterval.intersect` in `SpectroscopyModesMatrix`.

**Pull request merged**